test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs

```diff
@@ -21,6 +21,26 @@ public RazorPagesTest(MvcTestFixture<RazorPagesWebSite.Startup> fixture)
         public HttpClient Client { get; }
 
         [Fact]
+        public async Task Page_Handler_FormActionFromQueryString()
+        {
+            // Arrange & Act
+            var content = await Client.GetStringAsync("http://localhost/HandlerTestPage?formaction=Customer");
+
+            // Assert
+            Assert.StartsWith("Method: OnGetCustomer", content.Trim());
+        }
+
+        [Fact]
+        public async Task Page_Handler_FormActionRouteDataChosenOverQueryString()
+        {
+            // Arrange & Act
+            var content = await Client.GetStringAsync("http://localhost/HandlerTestPage/Customer?formaction=ViewCustomer");
+
+            // Assert
+            Assert.StartsWith("Method: OnGetCustomer", content.Trim());
```

Comment:
The output of OnGetCustomer and OnGetViewCustomerAsync would definitly be diferent if that's what you're asking?